bash_completion

```diff
@@ -623,7 +623,8 @@ _filedir()
         # the fallback condition with the "plus" dirs.
         local opts=(-f -X "$xspec")
         [[ $xspec ]] && plusdirs=(-o plusdirs)
-        [[ ${COMP_FILEDIR_FALLBACK-} || -z ${plusdirs-} ]] ||
+        [[ ${BASH_COMPLETION_FILEDIR_FALLBACK-${COMP_FILEDIR_FALLBACK-}} ||
+            -z ${plusdirs-} ]] ||
```

Comment:
```suggestion
            ! ${plusdirs-} ]] ||
```

It seems there are more cases `[[ ! $word ]]` (5 cases in `bash_completion`) than `[[ -z $word ]]` (2 cases). Another instance of `-z` is in line `bash_completion:1990` (`_cd`).